backend/src/core/services/tesseract.service.ts

```diff
@@ -0,0 +1,19 @@
+// import config from '@core/config'
+
+const VAULT_DOMAIN = 'storage.vault.gov.sg'
+
+/**
+ * Validate hash for a given campaignId, recipient and version
+ */
+const validateVaultUrl = (url: string): void => {
```

Comment:
Maybe this can be a separate Express middleware that responds with a 400 if the URL is invalid, and assigns the normalized URL to `res.locals.url` otherwise?